### PR TITLE
Also pass the dimension in which ASPECT will be run to the signals.

### PR DESCRIPTION
**Pull request by bangerth**
@jperryhouts -- this is for you.

### CODE REVIEW

**Comment by jperryhouts** [comment]
Cool. Merge this before my cookbook thing, and I'll update that PR to use this feature.

**Comment by bangerth** [comment]
Fixes #437

**Comment by tjhei** [comment]
:+1:

File: tests/signals.cc [hunk 1]
```diff
@@ -8,7 +8,8 @@ using namespace aspect;
 
 // create a function that is run upon loading the plugin
 // when declaring parameters, and that produces some output
-void declare_parameters(ParameterHandler &prm)
+void declare_parameters(const unsigned int dim,
+			ParameterHandler &prm)
```
Comment (gassmoeller) [inline tests/signals.cc]: This line should be indented by one more space, right?

File: source/simulator/simulator_signals.cc [hunk 2]
```diff
@@ -28,7 +28,8 @@ namespace aspect
 {
   // provide locations for the static variables of this class
   template <int dim>
-  boost::signals2::signal<void (ParameterHandler &)>  SimulatorSignals<dim>::declare_additional_parameters;
+  boost::signals2::signal<void (const unsigned int,
```
Comment (gassmoeller) [inline source/simulator/simulator_signals.cc]: Maybe I do not understand the structure behind these signals, but if we have `<int dim>` as template argument, why do we need another unsigned int as argument to this function?

**Comment by gassmoeller** [comment]
Apart from the things that I do not understand this looks fine. Timo you can merge this if you think it is correct, but I am a bit hesitant to merge stuff that I do not fully understand.

**Comment by jperryhouts** [comment]
Good question. I was under the impression that the declare_parameters signal is executed before the Simulator object is created, so aspect does not know the dimension yet, but if we can pass dim as a function argument, I don't know why we can't use it as a template argument. Doesn't make much difference to the implementation of sockets, though.

File: tests/signals.cc (hunk 1 above)
Comment (bangerth) [inline tests/signals.cc]: It's only in tests/ where the indentation script doesn't look. But I fixed it anyway.

File: source/simulator/simulator_signals.cc (hunk 2 above)
Comment (bangerth) [inline source/simulator/simulator_signals.cc]: It's a member function in a structure `SimulatorSignals<dim>` that depends on the dimension. The member function stores a pointer to the function. If you call the function pointed to, it doesn't know where you called it from, so the dimension is lost unless you specifically pass it as an argument.

**Pull request merged**